analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.vb

```diff
@@ -0,0 +1,503 @@
+        Return Not b = b AndAlso b.Value    ' FN, b.Value reached only when b is empty
+    End Function
+End Class
+
+Class ComplexConditionMultipleNullables
+    Private Function IndependentConditions1(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return f Is Nothing AndAlso d.Value = 42.0
+    End Function
+    Private Function IndependentConditions2(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return f IsNot Nothing AndAlso d.Value = 42.0
+    End Function
+    Private Function IndependentConditions3(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return f.HasValue AndAlso d.Value = 42.0
+    End Function
+    Private Function IndependentConditions4(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return f Is Nothing AndAlso d.Value = 42.0
+    End Function
+    Private Function IndependentConditions5(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return f Is Nothing AndAlso d.Value = 42.0
+    End Function
+
+    Private Function DependentConditions1(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return Not d.HasValue AndAlso d.Value = 42.0                                    ' Noncompliant
+    End Function
+    Private Function DependentConditions2(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return d.Value = 42.0 AndAlso d.HasValue                                        ' Compliant
+    End Function
+    Private Function DependentConditions3(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return d.Value = 42.0 AndAlso d.HasValue = f.HasValue AndAlso f.Value = 42.0    ' Compliant
+    End Function
+    Private Function DependentConditions4(ByVal d As Double?, ByVal f As Single?) As Boolean
+        Return d.Value = 42.0 AndAlso d.HasValue <> f.HasValue AndAlso f.Value = 42.0   ' Noncompliant
+    End Function
+
+    Private Sub ThirdExcluded(ByVal b1 As Boolean?, ByVal b2 As Boolean?)
+        If b1 = b2 AndAlso b1 <> b2 Then
+            Dim x = (CTypeDynamic(Of Integer?)(Nothing)).Value                          ' FIXME Non-compliant
+        End If
+    End Sub
+
+    Private Sub Transitivity(ByVal b1 As Boolean?, ByVal b2 As Boolean?, ByVal b3 As Boolean?)
+        If b1 = b2 AndAlso b1 <> b3 AndAlso b2 = b3 Then
+            Dim x = (CTypeDynamic(Of Integer?)(Nothing)).Value                          ' FIXME Non-compliant
+        End If
+
+        If b1 <> b2 AndAlso b1 <> b3 AndAlso b2 <> b3 AndAlso b1 IsNot Nothing AndAlso b2 IsNot Nothing Then
+            Dim x = b3.Value                                                            ' FN: b3 is empty
+        End If
+    End Sub
+
+    Private Sub Reachability1()
+        Dim b1 As Boolean? = Nothing
+        Dim x = b1.Value                ' Noncompliant
+        x = b1.Value                    ' Compliant, unreachable
+    End Sub
+
+    Private Sub Reachability2(ByVal b1 As Boolean?)
+        Dim b2 As Boolean? = Nothing
+
```

Comment:
As for cases above: here there is an empty line following `Dim`, which is inconsistent w.r.t. to the cases below (and some above too): e.g. https://github.com/SonarSource/sonar-dotnet/pull/6934/files#diff-d82d42eafc6dead52c4b51408059e1826011b06a45f9d16086cebe88438790adR307